lib/micro/case/utils.rb

```diff
@@ -16,17 +18,29 @@ def self.symbolize_keys(hash)
       end
     end
 
-    def self.stringify_keys(hash)
+    def stringify_keys(hash)
       return hash.transform_keys(&:to_s) if respond_to?(hash, :transform_keys)
 
       hash.each_with_object({}) { |(k, v), memo| memo[k.to_s] = v }
     end
 
-    def self.slice(hash, keys)
+    def slice(hash, keys)
       return hash.slice(*keys) if respond_to?(hash, :slice)
 
       hash.select { |key, _value| keys.include?(key) }
     end
+
+    def fetch_values(hash, keys, &block)
+      return hash.fetch_values(*keys, &block) if respond_to?(hash, :fetch_values)
+
+      result = []
+
+      keys.each do |key|
+        result << hash.fetch(key, &block)
+      end
+
+      result
+    end
```

Comment:
what do you think, if we use `each_with_object` to achieve the same result? 

```suggestion
      keys.each_with_object([]) do |key, values|
        values << hash.fetch(key, &block)
      end
    end
```

Reply:
Yes, it's a pretty nice idea, I think that this is a pretty good use case for `each_with_object` 😄